### ISSUE TITLE
Problem with the types : hybrid and structure

### ISSUE THREAD

**Issue by crawler-IM** [openpreserve/pagelyzer#6]
Hi,

We were testing the tool on 449 pairs and we encoutered some problems with the types hybrid and structure.
We should precise that the type=images has finished the comparaison successfully on 449 pairs.

Using hybrid and structure types, the pagelyzer_changedetection command calculate the score of the first 4 pairs and then it generates timeout errors (10 pairs of links are timeout) and finally the command crashs displaying this error :

/usr/lib/ruby/1.9.1/pagelyzer_util.rb:362:in `[]=': can't convert Symbol into String (TypeError)
        from /usr/lib/ruby/1.9.1/pagelyzer_util.rb:362:in`change_relative_url'
        from /usr/lib/ruby/1.9.1/pagelyzer_util.rb:372:in `block in change_relative_url'
        from /usr/lib/ruby/1.9.1/nokogiri/xml/node_set.rb:207:in`block in each'
        from /usr/lib/ruby/1.9.1/nokogiri/xml/node_set.rb:206:in `upto'
        from /usr/lib/ruby/1.9.1/nokogiri/xml/node_set.rb:206:in`each'
        from /usr/lib/ruby/1.9.1/pagelyzer_util.rb:371:in `change_relative_url'
        from /usr/lib/ruby/1.9.1/pagelyzer_util.rb:372:in`block in change_relative_url'
        from /usr/lib/ruby/1.9.1/nokogiri/xml/node_set.rb:207:in `block in each'
        from /usr/lib/ruby/1.9.1/nokogiri/xml/node_set.rb:206:in`upto'
        from /usr/lib/ruby/1.9.1/nokogiri/xml/node_set.rb:206:in `each'
        from /usr/lib/ruby/1.9.1/pagelyzer_util.rb:371:in`change_relative_url'
        from /usr/lib/ruby/1.9.1/pagelyzer_util.rb:372:in `block in change_relative_url'
        from /usr/lib/ruby/1.9.1/nokogiri/xml/node_set.rb:207:in`block in each'
        from /usr/lib/ruby/1.9.1/nokogiri/xml/node_set.rb:206:in `upto'
        from /usr/lib/ruby/1.9.1/nokogiri/xml/node_set.rb:206:in`each'
        from /usr/lib/ruby/1.9.1/pagelyzer_util.rb:371:in `change_relative_url'
        from /usr/lib/ruby/1.9.1/pagelyzer_util.rb:372:in`block in change_relative_url'
        from /usr/lib/ruby/1.9.1/nokogiri/xml/node_set.rb:207:in `block in each'
        from /usr/lib/ruby/1.9.1/nokogiri/xml/node_set.rb:206:in`upto'
        from /usr/lib/ruby/1.9.1/nokogiri/xml/node_set.rb:206:in `each'
        from /usr/lib/ruby/1.9.1/pagelyzer_util.rb:371:in`change_relative_url'
        from /usr/lib/ruby/1.9.1/pagelyzer_util.rb:389:in `block in normalize_DOM'
        from /usr/lib/ruby/1.9.1/nokogiri/xml/node_set.rb:207:in`block in each'
        from /usr/lib/ruby/1.9.1/nokogiri/xml/node_set.rb:206:in `upto'
        from /usr/lib/ruby/1.9.1/nokogiri/xml/node_set.rb:206:in`each'
        from /usr/lib/ruby/1.9.1/pagelyzer_util.rb:379:in `normalize_DOM'
        from /usr/lib/ruby/1.9.1/pagelyzer_analyzer.rb:258:in`start'
        from /usr/bin/pagelyzer_changedetection:298:in `block in <main>'
        from /usr/bin/pagelyzer_changedetection:250:in`each'
        from /usr/bin/pagelyzer_changedetection:250:in `<main>' 

Same thing using the option --timeout=120

Regards

**Comment by asanoja** [comment]
Hi,

Just to be precise I to see if I understand well the situation, in structural and hybrid:
4 pairs worked well
10 gives timeout
435 give this error described above

or just stop?

In a more technical detail, in fact, there is a bug that was not raise in our test, in the segmentation program to handle some kind of urls, for ViDIFF file generation porpouse. That is a minor change, it will be solved and commited to the repository, as soon as possible.

**Comment by crawler-IM** [comment]
No it crashs after the timeout errors : 
1) 4 pairs worked
2) 10 timeout
3) execution stopped dispaying that error (once) 

Thanks

**Comment by asanoja** [comment]
Ok, thanks for making it clear. Still, apply the same comment I made before. I'll notice you when the change is uploaded

**Comment by asanoja** [comment]
After been reviewed the code, there was a issue in the way intermediary data structure were generated in order to produce a ViXML structure, used in documents comparison. At first we have tried to force all urls into a document to be absolute, when in fact in some context this is not really needed. This feature has been disabled and we leave this task for after processed if needed by client programs.

When found urls in the document following the form: http://archive/path/http://archivedserver/archivedpath thrown this error. Eventhough we process the xml data structure for preventing this to happen, as we said, it has been disabled.

We have tested this fix with the 449 pairs given by mail and it has been successful in our installation.

Andrés
